File: surf-cloud-core/src/main/java/dev/slne/surf/data/core/spring/SurfSpringBanner.java
package dev.slne.surf.data.core.spring;

import java.io.PrintStream;
import org.springframework.boot.Banner;
import org.springframework.boot.ansi.AnsiColor;
import org.springframework.boot.ansi.AnsiOutput;
import org.springframework.core.env.Environment;

public class SurfSpringBanner implements Banner {

  @Override
  public void printBanner(Environment environment, Class<?> sourceClass, PrintStream out) {
    out.println(AnsiOutput.toString(AnsiColor.GREEN, "Starting plugin with Spring Boot..."));
  }
}

File: surf-cloud-core/src/main/java/dev/slne/surf/cloud/core/spring/SurfSpringBanner.kt
package dev.slne.surf.cloud.core.spring

import dev.slne.surf.cloud.api.util.buildAnsiString
import org.springframework.boot.Banner
import org.springframework.boot.ansi.AnsiColor
import org.springframework.core.env.Environment
import java.io.PrintStream

class SurfSpringBanner : Banner {
    override fun printBanner(environment: Environment, sourceClass: Class<*>?, out: PrintStream) {
        out.println(
            buildAnsiString(
                AnsiColor.GREEN,
                "Starting plugin with Spring Boot..."
            )
        )
    }
}
